src/sg/flybot/pullable/schema.clj

```diff
@@ -52,6 +55,62 @@
        {:registry general-pattern-registry}
        ::pattern]))))
 
+(defn add-pattern
+  "Adds pattern info to the shcema element `el` when applicable."
+  [el]
+  (if (vector? el)
+    (let [[k v] el]
+      (cond
+        (some #{k} [:map :vector :sequential :set])
+        el
+
+       (and (vector? v) (sequential? (last v)))
+        el
+
+        :else
+        [k [:or ::var v]]))
+    el))
+
+(defn remove-malli-options
+  "Malli syntax contains optional maps such as [:map {:closed true} [:a :int]].
+   Removes the option map from the schema element `el`."
+  [el]
+  (if (and (vector? el) (map? (second el)))
+    (let [[a _ & r] el]
+      (vec (concat [a] r)))
+    el))
+
+(def pattern-registry
+  "Combinations of malli default registries and our custom registry."
+  (merge (m/default-schemas) (mu/schemas) general-pattern-registry))
+
+(defn merge-schemas
+  "Merges 2 schemas."
+  [sch1 sch2]
+  (m/deref
+   (m/schema
+    [:merge sch1 sch2]
+    {:registry pattern-registry})))
+
+(defn combine-data-pattern
+  "Enhances the given `data-schema` with the pattern schema.
+   The malli option maps are first removed to avoid being walked by the pattern.
+   Then the pattern schema is being combined with the data-schema.
+   Finally, the malli option maps are added back via merging with original data-schema."
+  [data-schema]
+  (merge-schemas 
+   data-schema
+   (->> data-schema
+        (postwalk remove-malli-options)
+        (postwalk add-pattern))))
```

Comment:
`m/walk` seems to be for property operations (the optional malli maps), I don't think it can help with modifying the schema values directly.

I removed the malli properties because so I can easily use `postwalk` without affecting the properties. There should be a better way to handle this I think.